### PR TITLE
Refactoring logic around single replica deployments

### PR DESCRIPTION
**Pull request by barrymars**

### CODE REVIEW

**Review by nabadger — APPROVED** [review]

File: charts/standard-application-stack/tests/deployment_single_replica_test.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      global.clusterEnv: qa
+      allowSingleReplica: true
+    asserts:
+      - equal:
+          path: metadata.annotations.[app.mintel.com/opa-allow-single-replica]
+          value: "true"
+  - it: Check allowSingleReplica doesn't alter strategy
+    set:
+      global.clusterEnv: qa
+      allowsingleReplica: true
```
Comment (nabadger) [inline charts/standard-application-stack/tests/deployment_single_replica_test.yaml:41]: type here `allowS`

File: charts/standard-application-stack/tests/deployment_single_replica_test.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+suite: Test Deployment Single Replica Logic
```
Comment (nabadger) [inline charts/standard-application-stack/tests/deployment_single_replica_test.yaml:1]: might want to test for the case where both are set to true

File: charts/standard-application-stack/tests/deployment_single_replica_test.yaml (hunk 1 above)
Comment (barrymars) [reply charts/standard-application-stack/tests/deployment_single_replica_test.yaml]: fixed

File: charts/standard-application-stack/tests/deployment_single_replica_test.yaml (hunk 2 above)
Comment (barrymars) [reply charts/standard-application-stack/tests/deployment_single_replica_test.yaml:1]: not sure what you'd test for this as they only really interact in that they set the some annotation

**Review by nabadger — APPROVED** [review]

**Pull request merged**